From 05564ac0c21e99d5e1f3ad59b4a0281073b83c80 Mon Sep 17 00:00:00 2001
From: Ido Yariv <ido@wizery.com>
Date: Mon, 23 Apr 2012 19:10:51 +0300
Subject: [PATCH 19/19] mmc: omap_hsmmc: Avoid spurious interrupts in dat1
 handler

*** HACK ***

For some reason, probably because we request/free the interrupt handler each
time, we sometimes get spurious interrupts.
Some SDIO devices may enter low power mode. In this case, it might not be
possible to read the pending interrupts register, as the SDIO layer might not
be fully operational.
Check the gpio line before actually handling the interrupt.

TODO:
- Find out the root cause for the spruious interrupts
- Consider enabling/disabling the interrupt handler instead of
  requesting/freeing. This should be done regardless of this specific problem.

Signed-off-by: Ido Yariv <ido@wizery.com>
---
 drivers/mmc/host/omap_hsmmc.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/drivers/mmc/host/omap_hsmmc.c b/drivers/mmc/host/omap_hsmmc.c
index 0d6f082..08987c6 100644
--- a/drivers/mmc/host/omap_hsmmc.c
+++ b/drivers/mmc/host/omap_hsmmc.c
@@ -1349,6 +1349,10 @@ static irqreturn_t omap_hsmmc_dat1_irq(int irq, void *dev_id)
 {
 	struct omap_hsmmc_host *host = dev_id;
 
+	/* Avoid spurious interrupts */
+	if (gpio_get_value(mmc_slot(host).gpio_dat1))
+		return IRQ_HANDLED;
+
 	disable_irq_nosync(mmc_slot(host).dat1_irq);
 	mmc_signal_sdio_irq(host->mmc);
 
-- 
1.7.7.6

